PickNikRobotics/moveit_studio_example_behaviors
# Image behaviors

This PR adds two behaviors with the intention of demonstrating the following functionality:

- How to get a message from a ROS 2 topic using wait set
- How to put a ROS 2 message on the blackboard
  - Uses a sensor_msgs::msg::Image::SharedPtr instead of the message contents
- How to pass error messages to the UI (and a few instances of how to catch errors)
- How to get and use a ROS 2 message pointer from the blackboard
- How to write integration tests that include publishing ROS 2 messages

As well, the following functionality is also demonstrated:

- Converting sensor_msgs::msg::Image::SharedPtr to a cv_bridge::CvImagePtr
- Saving an image to disk
- Additional tests

Note: The ImageToBlackboard behavior has a doHalt(), which doesn't actually do anything significant but I wanted to include in one of the behaviors so the user can see how to use it, and (if they're paying attention) notice it is not necessary because it has not been included in the BlackboardImageToFile behavior
